### PR TITLE
Unit test for SecurityTokenManager.

### PR DESCRIPTION
**Pull request by imcarolwang**
For #3852 , and class name fix for SecurityTokenSerializerTest.cs.

@StephenBonikowsky, @mconnew, @HongGit

### CODE REVIEW

File: src/System.ServiceModel.Primitives/tests/IdentityModel/SecurityTokenSerializerTest.cs [hunk 1]
```diff
@@ -22,8 +22,8 @@ public static void Methods_NonNullParam_InvokeAndReturn()
         var xmlWriter = XmlWriter.Create(new MemoryStream());
         var dummyToken = new DummySecurityToken();
         var keyIdentifier = new SecurityKeyIdentifier();
-        var keyIdentifierClause = new SecurityKeyIdentifierClauseImp("DummyClause");
-        var sts = new SecurityTokenSerializerImp();
+        var keyIdentifierClause = new SecurityKeyIdentifierClauseImpl("DummyClause");
```
Comment (StephenBonikowsky) [inline src/System.ServiceModel.Primitives/tests/IdentityModel/SecurityTokenSerializerTest.cs]: Nice catching the typo :)

**Comment by imcarolwang** [comment]
@StephenBonikowsky  thank you for the guidance to fix the check error! I have run the rebase and push commands, and I got error "push reject due to current branch is behind its remote counterpart ", then I run a "pull" command and push again, it succeed. Now all the checks passed. But I see the changed files for the PR inclued more (some are merged already). I am not sure if that would affect final merge. Could you please help double check?

**Comment by StephenBonikowsky** [comment]
@imcarolwang the error you got when attempting to push was expected, basically we know that was was on your `https://github.com/imcarolwang` issue3852 branch was out-of-date. So after doing the rebase your local issue3852 branch was perfect. When attempting to push you got that error because your `https://github.com/imcarolwang` issue3852 branch thinks you were trying to push some older stuff to it. At that point you needed to do a force push...

`git push origin +issue3852`

When you did a pull at that point you pulled from `https://github.com/imcarolwang` issue3852 branch which then diffed to your local issue3852 and came up with what you see now.You don't want to merge this because since you did this we have already had an auto update #4013 and your changes would be reverting some of that.

If you do the same steps again but this time use the force push command it should be correct.

**Comment by imcarolwang** [comment]
Thanks @StephenBonikowsky! Everythong looks in a good state now :)

**Pull request merged**